dashboard/app/reporting_email.go

```diff
@@ -102,6 +106,113 @@ func (cfg *EmailConfig) Validate() error {
 	return nil
 }
 
+// handleCoverageReports sends reports for the previous period.
+// Assuming it is called June 15, the monthly report will cover April-May diff.
+func handleCoverageReports(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	curHostPort := r.URL.Host
+	targetDate := civil.DateOf(timeNow(ctx)).AddMonths(-1)
+	periods, err := coveragedb.GenNPeriodsTill(2, targetDate, "month")
+	if err != nil {
+		msg := fmt.Sprintf("error generating coverage report: %s", err.Error())
+		log.Errorf(ctx, "%s", msg)
+		http.Error(w, "%s: %w", http.StatusBadRequest)
+		return
+	}
+	wg := sync.WaitGroup{}
+	for nsName, nsConfig := range getConfig(ctx).Namespaces {
+		emailTo := nsConfig.Coverage.EmailRegressionsTo
+		minDrop := nsConfig.Coverage.RegressionThreshold
+		if emailTo != "" {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				if err := sendNsCoverageReport(ctx, nsName, emailTo, curHostPort, periods, minDrop); err != nil {
+					msg := fmt.Sprintf("error generating coverage report for ns '%s': %s", nsName, err.Error())
+					log.Errorf(ctx, "%s", msg)
+					w.Write([]byte(msg + "\n"))
+					return
+				}
+				w.Write([]byte(fmt.Sprintf("sent ns '%s' report to '%s'\n", nsName, emailTo)))
```

Comment:
We won't see these replies for any prod invocations, right? Perhaps better to write to log if that is something non-obvious.